src/tests/krate.rs

```diff
@@ -1447,6 +1447,251 @@ fn yank_max_version() {
+        assert_eq!(json.krate.documentation, Some("http://foo.rs".to_owned()));
+    }
+
+    // 3. Remove the documentation
+    {
+        let mut req = ::new_req_with_documentation(app.clone(), "docscrate", "0.2.2", None);
+        ::sign_in_as(&mut req, &user);
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: GoodCrate = ::json(&mut response);
+        assert_eq!(json.krate.documentation, None);
+    }
+
+    // Ensure latest version no longer has documentation
+    {
+        let mut req = ::req(app.clone(), Method::Get, "/api/v1/crates/docscrate");
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: CrateResponse = ::json(&mut response);
+        assert_eq!(json.krate.documentation, None);
+    }
+
+    // 4. Add documentation again to make sure this change isn't permanent
+    {
+        let mut req = ::new_req_with_documentation(
+            app.clone(),
+            "docscrate",
+            "0.2.3",
+            Some("http://bar.rs".to_owned()),
+        );
+        ::sign_in_as(&mut req, &user);
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: GoodCrate = ::json(&mut response);
+        assert_eq!(json.krate.documentation, Some("http://bar.rs".to_owned()));
+    }
+
+    // Ensure latest version also has the same documentation
+    {
+        let mut req = ::req(app.clone(), Method::Get, "/api/v1/crates/docscrate");
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: CrateResponse = ::json(&mut response);
+        assert_eq!(json.krate.documentation, Some("http://bar.rs".to_owned()));
+    }
+
+    // 5. Finally remove the documentation one more time just in case
+    {
+        let mut req = ::new_req_with_documentation(app.clone(), "docscrate", "0.2.4", None);
+        ::sign_in_as(&mut req, &user);
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: GoodCrate = ::json(&mut response);
+        assert_eq!(json.krate.documentation, None);
+    }
+
+    // Ensure latest version no longer has documentation
+    {
+        let mut req = ::req(app.clone(), Method::Get, "/api/v1/crates/docscrate");
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: CrateResponse = ::json(&mut response);
+        assert_eq!(json.krate.documentation, None);
+    }
+}
+
+#[test]
+#[ignore]
+fn publish_after_removing_documentation() {
+    let (_b, app, middle) = ::app();
+
+    let user;
+
+    // 1. Start with a crate with no documentation
+    {
+        let conn = app.diesel_database.get().unwrap();
+        user = ::new_user("foo").create_or_update(&conn).unwrap();
+        ::CrateBuilder::new("docscrate", user.id)
+            .version("0.2.0")
+            .expect_build(&conn);
+    }
+
+    // Verify that crates start without any documentation so the next assertion can *prove*
+    // that it was the one that added the documentation
```

Comment:
Similarly, this block (L1568-1575) tests the same thing that the previous test does in L1464-1471. I can't think of a scenario where one of them would pass and the other would fail. Could you remove this block please?